openpype/hosts/houdini/api/creator_node_shelves.py

```diff
@@ -59,9 +59,9 @@ def create_interactive(creator_identifier, **kwargs):
     """
 
     # TODO Use Qt instead
```

Comment:
Have we considered doing it like this instead?

```python

    host = registered_host()
    context = CreateContext(host)
    creator = context.manual_creators.get(creator_identifier)
    if not creator:
        raise RuntimeError("Invalid creator identifier: "
                           "{}".format(creator_identifier))

    # TODO Use Qt instead
    result, variant = hou.ui.readInput("Define variant name",
                                       buttons=("Ok", "Cancel"),
                                       initial_contents=creator.get_default_variant(),
                                       title="Define variant",
                                       help="Set the variant for the "
                                            "publish instance",
                                       close_choice=1)
    if result == 1:
        # User interrupted
        return
    variant = variant.strip()
    if not variant:
        raise RuntimeError("Empty variant value entered.")
```

That way we don't need to pass the argument plus the default variant is "live" to the creator (and thus from settings?) instead of being initialized once on Houdini launch. Not sure if better, might also be slightly slower when the node creation is actually triggered since it'll need to query the default variant from the creator at that moment in time.

Reply:
You are the one who wrote most of this so I would ask you if you ever consider it :)

But I don't see that many benefits on having it "live"?

Reply:
yeah I don't mind either TBH, I have done the changes now here https://github.com/ynput/OpenPype/pull/5421/commits/ab3d94fdb6f607b958704a06d3fa87187ac03d71 do you like it more?